try:
    a=[1,2,3]
    print(a[100])
except IndexError as e:
    print(e)
except NameError as n:
    print(n)
else:
    print("没有异常发生")
print('after except')

try:
    a=1/0
#不知道会出现哪种异常，可以采用以下方法捕捉异常，但是不推荐
except:
    print("出现异常了")


def divide(x,y):
    if y==0:
        raise Exception("divide zero except")
    return x/y

try:
    divide(19,0)
except Exception as e:
    print(e)